core/trashcan.ts

```diff
@@ -275,7 +275,13 @@ export class Trashcan extends DeleteArea implements IAutoHideable,
     const contents = this.contents_.map(function(string) {
       return JSON.parse(string);
     });
-    this.flyout?.show(contents);
+    // Trashcans with lots of blocks can take a second to render.
+    const blocklyStyle = this.workspace.getParentSvg().style;
+    blocklyStyle.cursor = 'wait';
+    setTimeout(() => {
```

Comment:
I'm sure I'm missing something obvious here, but what's the purpose of the setTimeout? Is it to ensure the wait cursor takes effect/is shown before things get blocked by the call to show()?

Reply:
Yup, the browser doesn't update the cursor if it's busy doing other things.  The setTimeout allows the browser a chance to do UI housekeeping.  The bad news is that the minimum value will change between browsers and systems, meaning this is not 100% reliable.  The good news is that failure is not a big deal, in fact it is simply the status quo.